### PR TITLE
sequencer: improve dirty state report message (CRAFT-302)

### PR DESCRIPTION
**Pull request by cmatsuoka**
Clarify the message to include the dependency step name in the reason
why the step is being re-executed.

Signed-off-by: Claudio Matsuoka <claudio.matsuoka@canonical.com>

- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?

-----

### CODE REVIEW

File: craft_parts/state_manager/reports.py [hunk 1]
```diff
@@ -134,7 +134,9 @@ def reason(self) -> str:
             if reasons_count > 1 or len(self.changed_dependencies) > 1:
                 reasons.append("dependencies")
             else:
-                reasons.append(f"{self.changed_dependencies[0].part_name!r}")
+                part_name = self.changed_dependencies[0].part_name
+                step_name = self.changed_dependencies[0].step.name.lower()
+                reasons.append(f"part {part_name!r} {step_name}")
```
Comment (sergiusens) [inline craft_parts/state_manager/reports.py:139]: Is there anything else aside from "changed"?

```suggestion
                reasons.append(f"{step_name} for part {part_name!r}")
```

**Review by sergiusens — APPROVED** [review]
Looks better, I added another suggestion.

**Review by facundobatista — APPROVED** [review]
Nice!

File: craft_parts/state_manager/reports.py (hunk 1 above)
Comment (cmatsuoka) [reply craft_parts/state_manager/reports.py:139]: It's only that for now, we can improve it in future iterations.

**Pull request merged**